src/build.rs

```diff
@@ -463,3 +470,46 @@ pub fn build(
         HumanDuration(started.elapsed())
     );
 }
+
+fn exec(command: &str, args: &Vec<&str>,
+        dir: &Path, started_ts: Instant,
+        pb: &ProgressBar) {
+    debug!("Running {} {:?}", command, args);
+    pb.set_message(&format!("{} {}", command, args.join(" ")));
+    let output = Command::new(command)
+        .env_clear()
+        .args(args)
+        .current_dir(dir)
+        .output()
+        .unwrap_or_else(|e| {
+            pb.println(format!(" {} {} {}",
+                       FAIL,
+                       command, args.join(" ")));
+            error!("build failed: {}", e);
+            process::exit(1)
+        });
+
+    debug!("stdout: {}", String::from_utf8_lossy(&output.stdout));
+    debug!("stderr: {}", String::from_utf8_lossy(&output.stderr));
+
+    match output.status.success() {
+        true => {
+            pb.println(format!(" {} {} {} (done in {})",
+                       CHECKMARK,
+                       command, args.join(" "),
+                       HumanDuration(started_ts.elapsed())));
+        }
+        false => {
+            pb.println(format!(" {} {} {}",
+                       FAIL,
+                       command, args.join(" ")));
+        }
+    }
+}
+
+fn has_openssl(src_dir: &Path) -> bool {
+    // check that lib/crypto/SKIP doesn't exist,
+    // if it does it means something went wrong with OpenSSL
+    ! src_dir.join("./lib/crypto/SKIP").exists()
```

Comment:
Oh nice! Just doing the check after running `configure` :). I was going to do a way more complicated solution, haha, this is much better.